### ISSUE TITLE
apache-noscript not catching AH00126 errors

### ISSUE THREAD

**Issue by FabioPedretti** [fail2ban/fail2ban#2908]
### Environment:
- Fail2Ban version (including any possible distribution suffixes): 0.10.2-2~bpo9+1
- OS, including release name/version: Debian 9
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:
It would be nice if apache-noscript could catch such AH00126 errors that I see in apache logs:
```
[Sat Oct 31 11:04:06.105212 2020] [core:error] [pid 17408] [client X.Y.Z.W:55280] AH00126: Invalid URI in request GET /static/../../../a/../../../../etc/passwd HTTP/1.1
[Sat Oct 31 11:04:11.774210 2020] [core:error] [pid 17768] [client X.Y.Z.W:45788] AH00126: Invalid URI in request GET /.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/.%2e/var/www/html/index.html HTTP/1.1
[Sat Oct 31 11:04:18.002892 2020] [core:error] [pid 17708] [client X.Y.Z.W:47600] AH00126: Invalid URI in request GET /dana-na/../dana/html5acc/guacamole/../../../../../../etc/passwd?/dana/html5acc/guacamole/ HTTP/1.1
[Sat Oct 31 11:04:18.029970 2020] [core:error] [pid 17759] [client X.Y.Z.W:47630] AH00126: Invalid URI in request GET /node_modules/../../../../../etc/passwd HTTP/1.1
[Sat Oct 31 13:13:31.340774 2020] [core:error] [pid 10628] [client X.Y.Z.W:51434] AH00126: Invalid URI in request GET /./../../../../../../../../../../etc/passwd HTTP/1.1
[Sat Oct 31 13:13:31.869619 2020] [core:error] [pid 11161] [client X.Y.Z.W:58180] AH00126: Invalid URI in request GET /../../../../../../../../../../../../../etc/passwd HTTP/1.1, referer: x.y.z
```
Thanks

**Comment by sebres** [comment]
Strictly seen, this kind of messages are not belonging to `apache-noscript` filter, which is targeting different vector and matching rather known only scripts, specified with parameter `script`, see https://github.com/fail2ban/fail2ban/blob/cc64ef25f68240d30a6653a2fba3cb26d574cd65/config/filter.d/apache-noscript.conf#L20
Probably filter `apache-overflows` would be more "suitable" filter for that. Just at the moment it'd also not match that.
Simple change of `(?:(?:AH0013[456]: )` to `(?:(?:AH001[23][456]: )` could fix that there:
```diff
- failregex = ^%(_apache_error_client)s (?:(?:AH0013[456]: )?Invalid (method|URI) in request\b|(?:AH00565: )?request failed: URI too long \(longer than \d+\)|request failed: erroneous characters after protocol string:|(?:AH00566: )?request failed: invalid characters in URI\b)
+ failregex = ^%(_apache_error_client)s (?:(?:AH001[23][456]: )?Invalid (method|URI) in request\b|(?:AH00565: )?request failed: URI too long \(longer than \d+\)|request failed: erroneous characters after protocol string:|(?:AH00566: )?request failed: invalid characters in URI\b)
```

If you want to have it nevertheless in `apache-noscript`, you can overwrite the regexs in `jail.local`:
```ini
[apache-noscript]
prefregex = ^<_apache_error_client> (?:(?:AH0(?:01(?:28|30)|1(?:264|071)|2811): )?(?:(?:[Ff]ile|script|[Gg]ot) )|(?:AH00126: )?(?=Invalid\b))<F-CONTENT>.+</F-CONTENT>$
failregex = %(known/failregex)s
            ^Invalid URI in request GET
```
Or customize it in own local filter `filter.d/apache-noscript.local`.
Just in any case note, that because historically all `apache-*` filters are basically poor customizable, this would completely overwrite original `prefregex`, so some new rules provided in some future update could be lost (if prefregex gets extended), and filter would not match some new known failures later - you have to check it by yourself everytime fail2ban gets updated.

**Comment by FabioPedretti** [comment]
Thanks for the reply.
I could customize my config like you suggested, but if this may be useful to others, would be possible to include it in official fail2ban, in the most suitable filter?
Searching on the web I find others users noting fail2ban missing AH00126 errors.
Thanks.

**Comment by sebres** [comment]
Sure... thus fixed in 9df332fdef853c0ae27578d84beda3a1c41deee1.
